src/site/xdoc/server/config-users.xml

```diff
@@ -48,6 +48,8 @@
         <dd>Algorithm to has passwords. Supported password algorithm are: MD5, SHA-256, SHA-512, NONE</dd>
         <dt><strong>enableVirtualHosting</strong></dt>
         <dd>true (default) or false. Defines if the usernames must (true) or may not contain (false) a domain part (user@domain.tld).</dd>
+          <dt><strong>forbiddenCharactersInLocalParts</strong></dt>
```

Comment:
I don't understand how it solve the problem : given that the path separator is by-session, how can you configure that forbiddenCharacterBlabla ? And want are you doing of wrong record already in database ?

Reply:
Netty imap implementation of imap protocol enforces a static '.' configuration, as well as storeMailboxManager. 

Which means we may limit alternative uses of Imap library, but that perfectly works on the packaged james versions. 

That is the best solution we came up with when @rouazana and me spotted the "user name should not contain '.' problem" on Thursday.

Well, we can very well decide to remove that commit, but an unperfect protection is better than none, no?

Reply:
No, a partial solution is often worse than no solution. Either we forbid the use of '.' character everywhere or we make it fully configurable, but there we just require our users to understand James limitation and then put this xml configuration because it can't really want without. I don't understand what the user gains by having this configuration.

Reply:
I vote: 'inject forbidden char' rather than configuration then.

Note: magic constant duplication and clear dependency is not ok to me. Then that job should be done by ioc code

Reply:
That is implemented in https://github.com/linagora/james-project/pull/1101